packages/medusa/src/services/cart.ts

```diff
@@ -1635,43 +1636,41 @@ class CartService extends TransactionBaseService {
   }
 
   /**
-   * Sets a payment method for a cart.
+   * Sets a payment method for a cart as selected and create the session to the external provider.
    * @param cartId - the id of the cart to add payment method to
    * @param providerId - the id of the provider to be set to the cart
-   * @return result of update operation
    */
-  async setPaymentSession(cartId: string, providerId: string): Promise<Cart> {
+  async setPaymentSession(cartId: string, providerId: string): Promise<void> {
```

Comment:
thought: It might be dangerous to change the return type if it is relied upon somewhere. Is this is necessary change?

Reply:
The actually returned cart is a cart without any relation or totals. I am not sure what it would be used for? what do you think?

Reply:
Fine with keeping it out 👍